oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/MultithreadedTraverseWithSortStrategy.java

```diff
@@ -254,7 +258,9 @@ private void resumeFromPreviousState(List<LastModifiedRange> previousState,
             LastModifiedRange currentRange = previousState.get(i);
             LastModifiedRange nextRange = i < previousState.size() - 1 ? previousState.get(i+1) : null;
             if (nextRange != null && currentRange.checkOverlap(nextRange)) {
-                throw new IllegalStateException("Range overlap between " + currentRange + " and " + nextRange);
+                LastModifiedRange merged = currentRange.mergeWith(nextRange);
+                log.info("Range overlap between " + currentRange + " and " + nextRange + ". Using merged range " + merged);
+                currentRange = merged;
```

Comment:
Not sure if I understand the logic... If you merge the ranges, wouldn't you want to remove nextRange as well? Is that anyway done and so not necessary?

Reply:
yes, you're right :) This didn't surface up during tests because having the second range as well was just duplicating the downloads but duplicates are finally removed while merging. 
Fixed the code to skip the next range if merge is done.